src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -699,11 +703,6 @@ export default class Fms {
      * @param runwayModel {RunwayModel}
      */
     replaceDepartureProcedure(routeString, runwayModel) {
-        // this is the same procedure that is already set, no need to continue
-        if (this.hasLegWithRouteString(routeString)) {
-            return;
-        }
-
         if (this.departureRunwayModel.name !== runwayModel.name) {
             this.setDepartureRunway(runwayModel);
```

Comment:
are you intending to use recursion here?  when `fms.replaceDepartureProcedure()` is called with a different runway, we would get the following:
```
.replaceDepartureProcedure()
  - > .setDepartureRunway()
    -> . _regenerateSidLeg()
        -> .replaceDepartureProcedure()
[finish first call to .replaceDepartureProcedure()]
```